eZ/Publish/Core/FieldType/RichText/InternalLinkValidator.php

```diff
@@ -28,6 +31,34 @@ public function __construct(ContentService $contentService, LocationService $loc
         $this->locationService = $locationService;
     }
 
+    public function validateDocument(DOMDocument $xml)
```

Comment:
I'm not sure where is the best place to put logic responsible for extracting the ezlocation:// and ezcontent:// links form XML document and validating them. WDYT @andrerom @alongosz ?

Other option would be field type definition itself, but I feel this logic is big enough to be a separate service.

Reply:
Hmm, judging from a screenshot I expected some checking if target object exists (so some interaction with Url Persistence handler). However I don't see it here and I'm lost. Don't we validate it via xsl?

Reply:
@alongosz As I've said on Slack: checking that target object exists is done via `validate` method: https://github.com/adamwojs/ezpublish-kernel/blob/ff60c8dc7fa9e908c0666c98c499723307314c58/eZ/Publish/Core/FieldType/RichText/InternalLinkValidator.php#L77-L100

BTW: This class was not used anywhere until now 😉